File: Data Structures & Algorithms/first-missing-positive/submission-2.cpp
class Solution {
public:
    int firstMissingPositive(vector<int>& nums) {

       set<int> seen;

       for(auto& num : nums){
        seen.insert(num);
       }

       for (int i = 1; i<=nums.size(); i++){
        if(seen.count(i) <= 0){
            return i;
        }
       }
        return nums.size()+1;
    }
};

File: Data Structures & Algorithms/top-k-elements-in-list/submission-2.py
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        #O(n)
        frequency = {}
        topElements = [None]*k
        #produce frequency list
        for num in nums:
            if num not in frequency:
                frequency[num] = 1
            else:
                frequency[num] += 1

        for i in range(k):
            topElements[i] = (max(frequency, key=frequency.get))
            frequency.pop(max(frequency, key=frequency.get))
        return topElements

        

        
        